src/main/java/org/jenkinsci/plugins/durabletask/AgentInfo.java

```diff
@@ -0,0 +1,169 @@
+                os = OsType.WINDOWS;
+            } else {
+                String osName = System.getProperty("os.name");
+                if (osName.equalsIgnoreCase("linux")) {
+                    os = OsType.LINUX;
+                } else if (osName.equalsIgnoreCase("z/OS")) {
+                    os = OsType.ZOS;
+                } else if (osName.equalsIgnoreCase("FreeBSD")) {
+                    os = OsType.FREEBSD;
+                } else {
+                    os = OsType.UNKNOWN;
+                }
+            }
+
+            String arch = System.getProperty("os.arch");
+            String archType = "";
+            if (os == OsType.DARWIN) {
+                if (arch.contains("aarch") || arch.contains("arm")) {
+                    archType = "arm";
+                } else if (arch.contains("amd") || arch.contains("x86")) {
+                    archType = "amd";
+                } else {
+                    archType = NOT_SUPPORTED;
+                }
+            }
+
+            // Note: This will only determine the architecture bits of the JVM.
+            String bits = System.getProperty("sun.arch.data.model");
+            if (bits.equals("64") || bits.equals("32")) {
+                archType += bits;
+            } else {
+                archType += NOT_SUPPORTED;
+            }
+
+            boolean binaryCompatible;
+            if ((os == OsType.DARWIN) || (os == OsType.LINUX) || (os == OsType.WINDOWS) && !archType.contains(NOT_SUPPORTED)) {
+                binaryCompatible = true;
+            } else {
+                binaryCompatible = false;
+            }
+
+            String extension = "";
+            if (os == OsType.WINDOWS) {
+                extension = ".exe";
+            }
+
+            String binaryName = BINARY_PREFIX + binaryVersion + "_" + os.getNameForBinary() + "_" + archType + extension;
+            String binaryPath;
+            boolean isCached;
+            boolean cachingAvailable;
+            try {
+                Path cachePath = Paths.get(nodeRoot.toPath().toString(), CACHE_PATH);
+                Files.createDirectories(cachePath);
+                File binaryFile = new File(cachePath.toFile(), binaryName);
+                binaryPath = binaryFile.toPath().toString();
+                isCached = binaryFile.exists();
+                cachingAvailable = true;
+            } catch (Exception e) {
+                // when the jenkins agent cache path is not accessible
+                binaryPath = binaryName;
+                isCached = false;
+                cachingAvailable = false;
+            }
+            AgentInfo agentInfo = new AgentInfo(os, archType, binaryCompatible, binaryPath, cachingAvailable);
+            agentInfo.setBinaryAvailability(isCached);
+            return agentInfo;
+        }
+
+        @Override
+        public void checkRoles(RoleChecker roleChecker) throws SecurityException {
+
+        }
```

Comment:
I think this method being empty is a potential security issue (the agent would be able to execute this on the controller, which in this case might not be a big deal but better not to allow it). Instead of having the class implement `FileCallable`, switch to extending `MasterToSlaveFileCallable` and then delete this override.